pkgs/by-name/po/positron/package.nix

```diff
@@ -0,0 +1,116 @@
+{
+  lib,
+  _7zz,
+  alsa-lib,
+  autoPatchelfHook,
+  dpkg,
+  fetchurl,
+  gtk3,
+  libglvnd,
+  libxkbcommon,
+  makeShellWrapper,
+  mesa,
+  musl,
+  nss,
+  stdenv,
+  systemd,
+  xorg,
+}:
+let
+  pname = "positron";
+  version = "2024.09.0-77";
+in
+stdenv.mkDerivation {
+  inherit version pname;
+
+  src =
+    if (stdenv.isDarwin) then
+      fetchurl {
+        url = "https://github.com/posit-dev/positron/releases/download/${version}/Positron-${version}.dmg";
+        hash = "sha256-rrcCqCLFx7EemTNaK5tPqRtbt+s4vLGSselSe8cCW1Y=";
+      }
+    else
+      fetchurl {
+        url = "https://github.com/posit-dev/positron/releases/download/${version}/Positron-${version}.deb";
+        hash = "sha256-AemC/Vl/fzWr6V0HshNBtMyZuyO916Vytfo0kzfk1Hs=";
+      };
+
+  buildInputs = lib.optionals stdenv.hostPlatform.isLinux [
+    alsa-lib
+    dpkg
+    gtk3
+    libglvnd
+    libxkbcommon
+    makeShellWrapper
+    mesa
+    musl
+    nss
+    stdenv.cc.cc
+    xorg.libX11
+    xorg.libXcomposite
+    xorg.libXdamage
+    xorg.libxkbfile
+  ];
+
+  nativeBuildInputs =
+    lib.optionals stdenv.hostPlatform.isLinux [
+      autoPatchelfHook
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [ _7zz ];
+
+  runtimeDependencies = lib.optionals stdenv.hostPlatform.isLinux [
+    # Needed to fix the "Zygote could not fork" error.
+    (lib.getLib systemd)
+  ];
+
+  unpackPhase = lib.optionals stdenv.hostPlatform.isLinux ''
+    dpkg-deb --fsys-tarfile $src | tar -x --no-same-owner
+    mkdir -p $out/share
+  '';
+
+  installPhase =
+    if stdenv.hostPlatform.isDarwin then
+      ''
+        mkdir -p $out/Applications
```

Comment:
Together with @saghirb we made it work by adding these lines:

```
        mkdir -p $out/Applications
        mkdir -p $out/bin
		cp -r . $out/Applications/Positron.app
		ln -s "$out/Applications/Positron.app/Contents/Resources/app/bin/code" "$out/bin/positron"
```

just as for linux, it seems we need symlink the executable to $out/bin

Reply:
Thanks!